### PR TITLE
[Improvement](sink) optimization for parallel result sink

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
## Proposed changes
optimization for parallel result sink #36053

### CODE REVIEW

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

File: be/src/runtime/buffer_control_block.cpp [hunk 1]
```diff
@@ -290,18 +277,25 @@ void BufferControlBlock::cancel() {
...
+    _update_dependency();
 }
 
 void BufferControlBlock::_update_dependency() {
-    if (_batch_queue_empty || _buffer_rows < _buffer_limit || _is_cancelled) {
-        for (auto dependency : _result_sink_dependencys) {
-            dependency->set_ready();
+    if (_is_cancelled) {
+        for (auto it : _result_sink_dependencys) {
+            it.second->set_ready();
```
Comment (Gabriel39) [inline be/src/runtime/buffer_control_block.cpp:289]:
```suggestion
            it.second->set_always_ready();
```

**Review by Gabriel39 — DISMISSED** [review]

**Review by HappenLee — DISMISSED** [review]
LGTM

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run buildall

**Comment by BiteTheDDDDt** [comment]
run performance

**Review by HappenLee — APPROVED** [review]
LGTM

**Review by zclllyybb — APPROVED** [review]

**Pull request merged**